### DISCUSSION TITLE
split provisioning conf

### DISCUSSION THREAD

**Discussion by didib** [grafana/grafana#51533] [Platform]
**What would you like to be added**:

It would be nice if there were multiple provisioning conf paths, perhaps (a subset of?):

/etc/grafana/conf/provisioning/datasources
/etc/grafana/conf/provisioning/notifiers
/etc/grafana/conf/provisioning/dashboards
/usr/share/grafana/conf/provisioning/datasources
/usr/share/grafana/conf/provisioning/notifiers
/usr/share/grafana/conf/provisioning/dashboards
/var/lib/grafana/conf/provisioning/datasources
/var/lib/grafana/conf/provisioning/notifiers
/var/lib/grafana/conf/provisioning/dashboards

Each with its own conf var. I can try to come up with suitable names, if needed, although naming isn't my strong side... e.g.:
conf_provisioning_datasources_path
conf_provisioning_notifiers_path
conf_provisioning_dashboards_path
sys_... (or default_..., or static_..., etc.)
var_... (or runtime_..., etc.)

**Why is this needed**:

In order to allow flexibility in having both packaged, static data (stuff that you get from an installed package when you install it, and do not change manually later), user-configurable data (which is commonly in /etc, and there are systems that (are trying to) keep /usr read-only), etc. Not sure what's the exact use-case of keeping also in /var, but in principle this is for program-managed data (not manually user-managed).

I am writing this while working on integration of oVirt DWH and grafana. I'd like to be able to supply a datasource (which is changeable, as it contains credentials, so should be in /etc) and some dashboards (which should be static, thus should be in /usr). For my own need, it might be enough to just have everything in /etc, or have conf/provisioning there and make dashboards a symlink to /usr. But that's a big ugly, IMO.

**Comment by bergquist** [comment]
Having one setting per provisioning makes sense. It should introduce a new section in the grafana config. Ex
```ini
[provisioning]
datasource_path = "/etc/grafana/provisioning/datasources"
dashboard_path = "..."
```

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
